pkg/volume/downwardapi/downwardapi.go

```diff
@@ -262,6 +261,24 @@ func CollectData(items []v1.DownwardAPIVolumeFile, pod *v1.Pod, host volume.Volu
 	return data, utilerrors.NewAggregate(errlist)
 }
 
+// TODO: unify with Kubelet.podFieldSelectorRuntimeValue
+// podFieldSelectorRuntimeValue extracts the field from the pod
+// and returns it as a string.
+func podFieldSelectorRuntimeValue(pod *v1.Pod, fieldPath string) (string, error) {
+	switch fieldPath {
+	case "spec.nodeName":
+		return pod.Spec.NodeName, nil
+	case "spec.serviceAccountName":
+		return pod.Spec.ServiceAccountName, nil
+	case "status.hostIP":
+		return pod.Status.HostIP, nil
```

Comment:
why does the kubelet provide this and pod IP directly rather than pulling from the pod object? I recall issues with order of operations and those particular values not being set in the pod when they were needed for downward API expansion